packages/core/src/Provider/StakePoolProvider/types/StakePoolProvider.ts

```diff
@@ -40,8 +49,14 @@ export interface QueryStakePoolsArgs {
 
     /** If provided, returns all the pools in any of the given status. */
     status?: Cardano.StakePoolStatus[];
+
+    /** Fuzzy search on text metadata fields. Ignored if shorter than 3 characters. If provided and valid, `identifier` is ignored. */
+    text?: string;
   };
 
+  /** Overrides default fuzzy options. Ignored in _live_ environments. */
+  fuzzyOptions?: FuzzyOptions;
```

Comment:
This might be a potential DoS vulnerability: making a lot of requests with slightly different fuzzy options weights or threshold would make it miss cache. Is this needed with current product requirements? If so, maybe we should separately cache [the result of all stake pool metadata query](https://github.com/input-output-hk/cardano-js-sdk/pull/1151/files#diff-a3d6cb4bc9a7a5e71b9eb05a20d21dde6e10155515c1a8769c2aad1ac5c66a2cR94) :thinking: 
```suggestion
  fuzzyOptions?: FuzzyOptions;
```

Reply:
> This might be a potential DoS vulnerability

Yes, indeed I added the option to disable this in `live-*` envs.

This is not a product requirement, I added the ability to override the options through API call as I found useful during development.

Next rather than removing it (as I thought somebody else may find it useful as I did) I tried to address the abuse attempts with the option.

Once said that, yes, caching separately query result and fuzzy index could be an interesting improvement quite cheap to be achieved: I'll do this.

Reply:
6deffdf484 should solve this